src/trunk.c

```diff
@@ -8964,39 +8965,52 @@ trunk_config_init(trunk_config        *trunk_cfg,
    trunk_cfg->max_pivot_keys = trunk_cfg->fanout + TRUNK_EXTRA_PIVOT_KEYS;
    uint64 header_bytes       = sizeof(trunk_hdr);
 
    uint64 pivot_bytes = (trunk_cfg->max_pivot_keys
                          * (data_cfg->max_key_size + sizeof(trunk_pivot_data)));
    uint64 branch_bytes =
       trunk_cfg->max_branches_per_node * sizeof(trunk_branch);
-   uint64 trunk_node_min_size = header_bytes + pivot_bytes + branch_bytes;
-   uint64 available_pivot_bytes =
-      cache_config_page_size(cache_cfg) - header_bytes - branch_bytes;
+   uint64 trunk_node_min_size   = header_bytes + pivot_bytes + branch_bytes;
+   uint64 page_size             = trunk_page_size(trunk_cfg);
+   uint64 available_pivot_bytes = page_size - header_bytes - branch_bytes;
    uint64 available_bytes_per_pivot =
       available_pivot_bytes / trunk_cfg->max_pivot_keys;
-   uint64 available_bytes_per_pivot_key =
-      available_bytes_per_pivot - sizeof(trunk_pivot_data);
-   platform_assert(trunk_node_min_size < cache_config_page_size(cache_cfg),
-                   "\nTrunk node does not fit in page size as configured.\n"
-                   "node->hdr: %luB\n"
-                   "pivots: %luB (max_pivot=%lu x %luB)\n"
-                   "branches %luB (max_branches=%lu x %luB).\n"
-                   "Maximum key size supported with current config: %luB.\n",
-                   header_bytes,
-                   pivot_bytes,
-                   trunk_cfg->max_pivot_keys,
-                   data_cfg->max_key_size + sizeof(trunk_pivot_data),
-                   branch_bytes,
-                   max_branches_per_node,
-                   sizeof(trunk_branch),
-                   available_bytes_per_pivot_key);
+
+   // Deal with mis-configurations where we don't have available bytes per
+   // pivot key
+   uint64 available_bytes_per_pivot_key = 0;
+   if (available_bytes_per_pivot > sizeof(trunk_pivot_data)) {
+      available_bytes_per_pivot_key =
+         available_bytes_per_pivot - sizeof(trunk_pivot_data);
+   }
+
+   if (trunk_node_min_size >= page_size) {
+      platform_error_log("Trunk node min size=%lu bytes "
+                         "does not fit in page size=%lu bytes as configured. "
+                         "node->hdr: %lu bytes, "
+                         "pivots: %lu bytes (max_pivot=%lu x %lu bytes), "
+                         "branches %lu bytes (max_branches=%lu x %lu bytes), "
+                         "Maximum key size supported with current "
+                         "configuration: %lu bytes.\n",
+                         trunk_node_min_size,
+                         page_size,
+                         header_bytes,
+                         pivot_bytes,
+                         trunk_cfg->max_pivot_keys,
+                         trunk_pivot_size,
+                         branch_bytes,
+                         max_branches_per_node,
+                         sizeof(trunk_branch),
+                         available_bytes_per_pivot_key);
```

Comment:
An example of this new error message when running `driver_test splinter_test` is here:

```
sdb-fdb-build:[65] $ ./build/debug/bin/driver_test splinter_test --functionality 1000000 100 --page-size 4096 --fanout 100
./build/debug/bin/driver_test: splinterdb_build_version d852a1db-dirty
Dispatch test splinter_test
Trunk node min size=13580 bytes does not fit in page size=4096 bytes as configured. node->hdr: 1198 bytes, pivots: 12190 bytes (max_pivot=106 x 115 bytes), branches 192 bytes (max_branches=24 x 8 bytes), Maximum key size supported with current configuration: 0 bytes.
Failed to parse config arguments. See 'driver_test splinter_test --help' for usage information: Invalid argument
```

Reply:
This is nice. Can you reformat it so as to print multiple 80 char lines?

Reply:
Changed as suggested.